### DISCUSSION TITLE
sys.implementation test for being a tuple fails.

### DISCUSSION THREAD

**Discussion by robert-hh** [micropython/micropython#10101] [Core Development]
The value of sys.implentation cannot be tested as being a tuple. That used to work, but does not any more. That may have been discussed already, bur I cannot recall that. Sample sequence:
```
>>> import sys
>>> s=sys.implementation
>>> type(s)
<class 'tuple'>
>>> type(s) is tuple
False
```
but:
```
>>> t=(1,2,3)
>>> type(t)
<class 'tuple'>
>>> type(t) is tuple
True

```

**Comment by dlech** [comment]
Since `sys.implementation` is required to have named attributes, it doesn't seem like it should be exactly a `tuple`. Does `isinstance(t, tuple)` work?

**Reply by robert-hh** [reply]
Nope:
```
>>> import sys
>>> s=sys.implementation
>>> isinstance(s, tuple)
False
>>> 
```
but:
```
>>> t=(1,2,3)
>>> isinstance(t, tuple)
True
>>> 
```

**Reply by dlech** [reply]
It looks like the type depends on `MICROPY_PY_ATTRTUPLE`, it could be `mp_type_tuple` or `mp_type_attrtuple`. But in any case it seems like, `type(t).__name__ == "tuple"` should work.

Why do you need to test the type?

**Reply by robert-hh** [reply]
`MICROPY_PY_ATTRTUPLE` is enabled. `type(t).__name__ == "tuple"` works.
Actually, I do not need it. I was just puzzled because is used to work a while ago and now it does not any more. I came along it when I tried to run Python code which I did not use since 2020.

**Reply by peterhinch** [reply]
I had assumed it was a `namedtuple` but my tests suggest it isn't (in MP or in CPython).

**Reply by jimmo** [reply]
Yeah, it's an `attrtuple` which cannot be created or accessed from Python, but (confusingly) shares the same name QSTR as regular tuple.

With attrtuple support disabled, then:

```
>>> type(sys.implementation) is tuple
True
```
